From 517d12fb632fad4bef5a957d6c83fa40cc99aba3 Mon Sep 17 00:00:00 2001
From: Will Dietz <will.dietz@sifive.com>
Date: Tue, 22 Aug 2023 11:40:47 -0500
Subject: [PATCH] mlir-opt: Fix linking after 7c4e8c6a273f2 .

Without this, undefined refernces to the LLVMIR translations:
```
ld: mlir-opt.cpp:(.text.startup.main+0x49): undefined reference to `mlir::registerAMXDialectTranslation(mlir::DialectRegistry&)'
ld: mlir-opt.cpp:(.text.startup.main+0x51): undefined reference to `mlir::registerArmSMEDialectTranslation(mlir::DialectRegistry&)'
ld: mlir-opt.cpp:(.text.startup.main+0x59): undefined reference to `mlir::registerArmSVEDialectTranslation(mlir::DialectRegistry&)'
ld: mlir-opt.cpp:(.text.startup.main+0x81): undefined reference to `mlir::registerOpenACCDialectTranslation(mlir::DialectRegistry&)'
ld: mlir-opt.cpp:(.text.startup.main+0x89): undefined reference to `mlir::registerOpenMPDialectTranslation(mlir::DialectRegistry&)'
ld: mlir-opt.cpp:(.text.startup.main+0x99): undefined reference to `mlir::registerX86VectorDialectTranslation(mlir::DialectRegistry&)'
```
---
 mlir/tools/mlir-opt/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mlir/tools/mlir-opt/CMakeLists.txt b/mlir/tools/mlir-opt/CMakeLists.txt
index df1e9480c754..ff4c08081334 100644
--- a/mlir/tools/mlir-opt/CMakeLists.txt
+++ b/mlir/tools/mlir-opt/CMakeLists.txt
@@ -5,6 +5,8 @@ set(LLVM_OPTIONAL_SOURCES
 get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
 get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
 get_property(extension_libs GLOBAL PROPERTY MLIR_EXTENSION_LIBS)
+get_property(translation_libs GLOBAL PROPERTY MLIR_TRANSLATION_LIBS)
+
 set(LLVM_LINK_COMPONENTS
   Core
   Support
@@ -53,6 +55,7 @@ set(LIBS
   ${dialect_libs}
   ${conversion_libs}
   ${extension_libs}
+  ${translation_libs}
   ${test_libs}
 
   MLIRAffineAnalysis
-- 
2.41.0

